pueue/src/daemon/service.rs

```diff
@@ -0,0 +1,639 @@
+use std::{
```

Comment:
> Explain a few simple things, such as how windows services work (they need to be registered and are afterwards permanently known to the system) and how are they started/stopped.

Services are installed by an administrator user. There is an official Windows service manager one can use to start/stop/restart the service, though we also included this functionality in the binary (minus the restart one, but this can be emulated by the user anyways with 2 commands). On install, it is set by default to autostart, and the system starts the service every time you start the system (unless the user changed the setting to manual or disabled in the service manager).

When a service is installed, Windows calls a preset cli string (our binary + args) which communicates with the service api (hence why we have a special flag for this, to trigger the api communication). We use an absolute path, so, the binary cannot be moved. This is to avoid the need to add it to PATH (which is an annoyance for the common user).

When the system is started, it listens for events as you've said, and when a user logs in, spawns a pueued daemon as the user, and on logoff it kills it (this technically is a noop. Windows itself kills all user processes on logoff anyways. More will be explained in the module docs)

> From what I understand, the service is registered with the args to call the service run subcommand whenever the service is to be started, which then listens to certain system level events, such as shutdown/logout/logon.
> 
> On start, that pueued instance then goes ahead and spawns another pueued instance, which is then the actual pueued instance.

Basically yes! (Adding module docs soon)

Reply:
Module docs added, let me know if it's sufficient for you

Reply:
Module docs added. Let me know if it's good enough for you!